matrix/src/dense.rs

```diff
@@ -550,6 +550,34 @@ impl<T: Clone + Default + Send + Sync> DenseMatrix<T> {
         assert!(new_height >= self.height());
         self.values.resize(self.width * new_height, fill);
     }
+
+    /// Pad the matrix height to the next power of two by appending rows filled with `fill`.
+    ///
+    /// This is commonly used in proof systems where trace matrices must have power-of-two heights.
+    ///
+    /// # Behavior
+    ///
+    /// - If the matrix is empty (height = 0), it is padded to have exactly one row of `fill` values.
+    /// - If the height is already a power of two, the matrix is unchanged.
+    /// - Otherwise, the matrix is padded to the next power of two height.
+    pub fn pad_to_power_of_two_height(&mut self, fill: T) {
+        // Handle empty matrix: ensure at least one row of fill values.
+        if self.height() == 0 {
+            self.values.resize(self.width, fill);
+            return;
+        }
+
+        // Compute the target height as the next power of two.
+        let target_height = self.height().next_power_of_two();
+
+        // If already a power of two, nothing to do.
+        if target_height == self.height() {
+            return;
+        }
+
+        // Pad with the specified fill value.
+        self.values.resize(self.width * target_height, fill);
```

Comment:
I think we can simplify the implementation here.

`0..next_power_of_two()` returns one so there is no need for the special case and if we switch from an equality if statement to an inequality we need several fewer lines of code.

Moreover, if `new_len = len` resize has no effect.
```suggestion
        // Compute the target height as the next power of two.
        let target_height = self.height().next_power_of_two();
        
        // If target_height == height, resize will have no effect.
        // Otherwise we pad the matrix to a power of two height by filling with the supplied value.
        self.values.resize(self.width * target_height, fill);
```